pyrightconfig.json

```diff
@@ -0,0 +1,30 @@
+{
+  "pythonVersion": "3.8",
+  "include": ["augur/"],
+
+
+  // These are rules that are enabled by default and currently have violations.
+
+  // TODO: go through these exceptions and determine if they should be kept or
+  // removed and violations addressed.
+
+  // TODO: consider aligning this with the same file in nextstrain/cli:
```

Comment:
I looked at a few of these - totally fine if we want to defer any and all to separate PRs beyond this one:
* `reportUnusedExpression` - single failure, should fix
* `reportUndefinedVariable` - single failure, already with a flake8 ignore comment
* `reportPrivateImportUsage` - no failures
* `reportGeneralTypeIssues` - 9 errors, but all seem worth fixing

Reply:
#1472